src/main/java/ru/investbook/ApplicationFailedRunListener.java

```diff
@@ -18,23 +18,27 @@
 
 package ru.investbook;
 
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
 import org.springframework.boot.context.event.ApplicationFailedEvent;
 import org.springframework.context.ApplicationListener;
 import org.springframework.core.env.ConfigurableEnvironment;
 
+@Slf4j
+@RequiredArgsConstructor
 public class ApplicationFailedRunListener implements ApplicationListener<ApplicationFailedEvent> {
 
-    private final Logger log = LoggerFactory.getLogger(ApplicationFailedRunListener.class);
+    private final LoadingPageServer loadingPageServer;
 
     @Override
     public void onApplicationEvent(ApplicationFailedEvent event) {
         ConfigurableEnvironment environment = event.getApplicationContext().getEnvironment();
         if (environment.getProperty("investbook.open-home-page-after-start", Boolean.class, false)) {
             log.info("Application run failed. May be application was run a second time, trying to open the application page...");
-            int port = environment.getProperty("server.port", Integer.class, 8080);
-            BrowserHomePageOpener.open("http://localhost:" + port);
+
+            if (loadingPageServer != null && !loadingPageServer.isRunning()) {
+                loadingPageServer.start();
+            }
```

Comment:
Now `loadingPageServer` is never GC-ed.

Please, create new instance inside `if (environment.getProperty("investbook.open-home-page-after-start", Boolean.class, false)) {` block and remove instance field.

Error is a rare case, we shouldn't hold error handing object in alive state when app is working.
```java
LoadingPageServer loadingPageServer = new LoadingPageServer();
loadingPageServer.start();
loadingPageServer.stopAfter(120);
```